test/utils/mochaHooks.test.js

```diff
@@ -64,19 +64,21 @@ describe('mochaHooks', () => {
           return null;
         });
 
-        let setState;
+        let unsafeSetState;
         function Parent() {
-          setState = React.useState(0)[1];
+          const [state, setState] = React.useState(0);
+          unsafeSetState = setState;
+
           React.useEffect(() => {});
           React.useEffect(() => {});
 
-          return <Child />;
+          return <Child rerender={state} />;
```

Comment:
[previous transpilation](https://babeljs.io/repl#?browsers=ie%2011&build=&builtIns=false&spec=false&loose=false&code_lz=MYewdgzgLgBAwgCwJYBsAmMC8MBKBTAQ2CgDoAzEAJwHcDK18yAKMgVzGKXHmXSYEoYAbwBQMcRMn4ipVhDwBRMmTzEmArAD5hAX34BuMZOPTiJOYuWqo6wZm1C9h48cp4orSmBhhWKFM4uME4iRi4o7jDyUADKUARQeIEubBxQXN4ACnR4YDaCokESoJCwANrQCXgANFHucVUAuli4hGYWDYlMAAz8ZQCMjYZhQaay8koqahr2ugYjLmPmE1bTdg4hC67unt4APIioGDAA9JrJxjpAA&debug=false&forceAllTransforms=false&shippedProposals=true&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=react&prettier=false&targets=&version=7.13.11&externalPlugins=%40babel%2Fplugin-transform-react-constant-elements%407.13.10)

This meant that React bailed out of re-rendering `<Child />` when calling `setState(1)` because `<Child />` was hoisted.